### ISSUE TITLE
Ingress rule created when the target pods are still not fully created had incorrect backend port configured.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#5825]
Rancher Version:v1.1.3

Steps to reproduce the problem:
Create k8s environment with 2 hosts.

Use the following yml file to create services and ingress at the same time:

```
apiVersion: v1
kind: Service
metadata:
  name: k8test5
  labels:
    k8s-app: k8test5-service
spec:
  ports:
  - port: 80
    targetPort: 80
    protocol: TCP
    name: http
  selector:
    k8s-app: k8test5-service
---
apiVersion: v1
kind: ReplicationController
metadata:
  name: k8test5
spec:
  replicas: 2
  selector:
    k8s-app: k8test5-service
  template:
    metadata:
      labels:
        k8s-app: k8test5-service
    spec:
      terminationGracePeriodSeconds: 60
      containers:
      - name: testcontainer
        image: sangeetha/testnewhostrouting
        ports:
        - containerPort: 80
        tty: true
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
 name: ingress5
 annotations:
   http.port: "87"
spec:
  backend:
    serviceName: k8test5
    servicePort: 80

```

When lb service comes up , it is not reachable.
haproxy configuration has incorrect backend port configured in this case.

<img width="756" alt="screen shot 2016-08-26 at 5 25 12 pm" src="https://cloud.githubusercontent.com/assets/4266958/18023794/6d33bfbc-6bb2-11e6-9c80-2eb0bab2a155.png">

Note  - I saw the same behavior when all the existing services and ingress rules was deleted and had to get recreated when etcd was restored from backup files.

**Comment by alena1108** [comment]
This issue should be fixed for both 1.2 and 1.3 k8s templates

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.1.4-rc1 and K8s template from https://github.com/rancher/qa-catalog.

This issue is not seen anymore. Lb ip adress redirects traffic to targets as expected.
